dist/gcc-compatible/Hacl_IntTypes_Intrinsics.h

```diff
@@ -42,10 +42,18 @@ extern "C" {
 static inline uint32_t
 Hacl_IntTypes_Intrinsics_add_carry_u32(uint32_t cin, uint32_t x, uint32_t y, uint32_t *r)
 {
-  uint32_t res = x + cin + y;
-  uint32_t
-  c = (~FStar_UInt32_gte_mask(res, x) | (FStar_UInt32_eq_mask(res, x) & cin)) & (uint32_t)1U;
-  r[0U] = res;
+  uint64_t res = (uint64_t)x + (uint64_t)cin + (uint64_t)y;
```

Comment:
is this faster, then?

Reply:
Some numbers again for both changes would be nice.

Reply:
As we discussed in the previous PR, we can't really measure these accurately on ARM at the moment but I can measure the P-256 functions and these are significantly faster. For example, on a small Odroid C4 board I get these speed-ups (in microseconds):
```
P-256/verify 13684 -> 7218
P-256/compressed_to_raw 271 -> 147
P-256/dh_initiator 4240 -> 2247
P-256/valid_pk 4224 -> 2233
```

I wanted to confirm these on an M1 Mac too as I did originally but I don't have access to one right now, I'll report when I will.

Reply:
I can confirm these speed-ups on M1 Mac Mini too:
```
P-256/P-256/verify 2542 -> 803
P-256/P-256/compressed_to_raw 50 -> 16
P-256/P-256/dh_initiator 785 -> 247
P-256/P-256/valid_pk 787 -> 248
```

Reply:
👏